react/utils/metrics/createQuote.ts

```diff
@@ -0,0 +1,135 @@
+import axios from 'axios'
+
+import type { Metric, SessionResponse } from './metrics'
+import { sendMetric } from './metrics'
+
+const GRAPHQL_URL = (accountName: string, workspace?: string) => {
+  if (workspace) {
+    return `https://${workspace}--${accountName}.myvtex.com/_v/private/graphql/v1`
+  }
+
+  return `https://${accountName}.myvtex.com/_v/private/graphql/v1`
+}
+
+type CreateQuoteFieldsMetric = {
+  cost_center_id: string
+  cost_center_name: string
+  buy_org_id: string
+  buy_org_name: string
+  member_id: string
+  member_email: string
+  role: string
+  creation_date: string
+  quote_id: string
+  quote_reference_name: string
+  send_to_sales_rep: boolean
+}
+
+type CreateQuoteMetricsParam = {
+  quoteId: string
+  sessionResponse: SessionResponse
+  workspace: string
+  account: string
+  sendToSalesRep: boolean
+}
+
+type CreateQuoteMetric = Metric & { fields: CreateQuoteFieldsMetric }
+
+const fetchMetricsData = async (
+  accountName: string,
+  workspace: string,
+  quoteId: string,
+  userEmail: string
+) => {
+  const query = JSON.stringify({
+    query: `query GetMetricsData($id: String, $email: String!) {
+      getQuote(id: $id) @context(provider: "vtex.b2b-quotes-graphql") {
+        organization
+        organizationName
+        costCenterName
+        referenceName
+        creatorRole
+        creationDate
+      },
+      getUserByEmail(email: $email) @context(provider: "vtex.storefront-permissions") {
+        costId
+    }
+  }`,
+    variables: { id: quoteId, email: userEmail },
+  })
+
+  const { data, errors } = (
+    await axios.post(GRAPHQL_URL(accountName, workspace), query)
+  ).data
+
+  if (errors) {
+    console.error('Graphql errors', errors)
+    throw new Error('Graphql Errors when trying get quote and user data')
+  }
+
+  const quoteResult = data?.getQuote as Omit<QuoteMetricsData, 'costId'>
+  const costId = (data?.getUserByEmail?.[0].costId ?? '') as string
+
+  return {
+    ...quoteResult,
+    costId,
+  }
+}
+
+const buildCreateQuoteMetric = async (
+  metricsParam: CreateQuoteMetricsParam
+): Promise<CreateQuoteMetric> => {
+  const { namespaces } = metricsParam.sessionResponse
+  const userEmail = namespaces?.profile?.email?.value
+
+  const metricsData = await fetchMetricsData(
+    metricsParam.account,
+    metricsParam.workspace,
+    metricsParam.quoteId,
+    userEmail
+  )
+
+  const metric: CreateQuoteMetric = {
+    name: 'b2b-suite-buyerorg-data',
+    kind: 'create-quote-ui-event',
+    description: 'Create Quotation Action - UI',
+    account: metricsParam.account,
+    fields: {
+      buy_org_id: metricsData.organization,
+      buy_org_name: metricsData.organizationName,
```

Comment:
I've just noticed that this PR also has the typos:

```suggestion
      buyer_org_id: metricsData.organization,
      buyer_org_name: metricsData.organizationName,
```

Can you check if there are others and fix them? Same for the metrics for using quotes.